lockbox.go

```diff
@@ -134,76 +145,88 @@ func (l *Lockbox) Lock() error {
 }
 
 // Unlock will unlock the given lockbox, as long as the provided key[s] are valid.
-func (l *Lockbox) Unlock(namespace, code, salt string) error {
-	if salt == "" {
-		return errors.New("salt is necessary to set the value")
-	}
-	enckey, err := l.GetMetaData(fmt.Sprintf("%v/otp/key", namespace))
-	// fmt.Println("ENCKEY SIZE:", len(enckey))
+func (l *Lockbox) Unlock(namespace, username, password, code string) error {
+	// get our encrypted otp key from the metadata store
+	encotpkey, err := l.GetMetaData(fmt.Sprintf("%v/otp/key", namespace))
 	if err != nil {
 		return err
 	}
-	securePassphrase := pbkdf2.Key([]byte(l.MasterKey), []byte(salt), 4096, 10, sha1.New)
-	keyurl, err := decrypt(enckey, string(securePassphrase))
+
+	user := &User{
+		Username: username,
+		Password: password,
+	}
```

Comment:
Everything here seems fine, the only thing that I think we can do better is the creation of the user. We can have a construction function (`NewUser`) which will take the only username and password. By doing that way we can also set the `EncryptionKey` at the time of instantiating.

Reply:
Im fine with adding a function but what advantage does that give us in this particular case?